core/domain/exp_services.py

```diff
@@ -926,16 +927,20 @@ def update_exploration(
             'Exploration is public so expected a commit message but '
             'received none.')
 
-    if (is_suggestion and (
-            not commit_message or
-            not commit_message.startswith(
-                feconf.COMMIT_MESSAGE_ACCEPTED_SUGGESTION_PREFIX))):
-        raise ValueError('Invalid commit message for suggestion.')
-    if (not is_suggestion and commit_message and commit_message.startswith(
-            feconf.COMMIT_MESSAGE_ACCEPTED_SUGGESTION_PREFIX)):
-        raise ValueError(
-            'Commit messages for non-suggestions may not start with \'%s\'' %
-            feconf.COMMIT_MESSAGE_ACCEPTED_SUGGESTION_PREFIX)
+    suggestion_commit_message_prefix_regex = '^%s' % (
+        feconf.COMMIT_MESSAGE_ACCEPTED_SUGGESTION_PREFIX_TEMPLATE.replace(
+            '%s', '.*'))
+    if is_suggestion:
+        if not commit_message or not re.match(
+                suggestion_commit_message_prefix_regex, commit_message):
```

Comment:
Are we retroactively updating earlier commit messages to match this? Otherwise, this seems to make data integrity a bit more difficult since we'll now have a different hard enforcement of formatting at different times in exploration histories (even two explorations of the same version may have different commit message formats).